### ISSUE TITLE
[Bug]: Can't authenticate with x5c provisioner to step ca admin

### ISSUE THREAD

**Issue by miguelangel-nubla** [smallstep/cli#860]
### Steps to Reproduce

```
pi@tinyca:~$ step ca admin list
No admin credentials found. You must login to execute admin commands.
✔ Please enter admin name/subject (e.g., name@example.com): step
✔ Provisioner: provisioner1 (X5C)
provisioner type 'X5C' requires the '--x5c-cert' flag
pi@tinyca:~$ step ca admin list --x5c-cert
Incorrect Usage: flag provided but not defined: -x5c-cert
```

### Your Environment

* `step` CLI Version - 0.23.2

### Expected Behavior

Authenticate using X5C credentials, as you can even create a Super Admin with that provisioner:
```
SUBJECT PROVISIONER     TYPE
step    admin (JWK)    SUPER_ADMIN
step    provisioner1 (X5C)      SUPER_ADMIN
```

### Actual Behavior

throws:
`provisioner type 'X5C' requires the '--x5c-cert' flag`

### Contributing

Vote on this issue by adding a 👍 reaction.
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by miguelangel-nubla** [comment]
Same happens with `step ca token`.
Is this by design?

**Comment by maraino** [comment]
cc @dopey

A workaround might be to create a certificate using that provisioner and then use it with `--admin-cert` and `--admin-key`

**Comment by miguelangel-nubla** [comment]
The use case is using hardware signing (yubikey or pkcs11) for as much as possible, as to not have the chance of a private key or passphrase leaking.

The ability to use a `--x5c-cert` and `--x5c-key` together with `--kms` to authenticate to a provisioner reduces the attack vector considerably. An attacker with full access to your system will be limited to the actions you have signed (ideally with the touch of the yubikey), nothing more, instead of full access to the scope of the provisioner.

It just works already with ssh certificates:
```
step ssh certificate \
--kms "yubikey:?pin-value=123456" \
--x5c-cert "yubikey:slot-id=85" \
--x5c-key "yubikey:slot-id=85" \
--provisioner provisioner1  user
```
which I think is awesome.

Would love to see support for it everywhere, other more useful examples that come to mind are `step ca token` and `step ssh login` with the _touch of a button_, and no higher level secrets at risk.

**Comment by maraino** [comment]
> Same happens with `step ca token`.
> Is this by design?

No, we want to support kms everywhere, but we need to dedicate time to do it, and we've only done it in the most commonly used commands.

Some time ago I created [this script](https://gist.github.com/maraino/f73b334e475caa1a0f34a52c501fede7) that you should be able to use with a YubiKey too.

Adding kms support for all the admin commands might be a bit hard, but adding it to `step ca token` and adding an `--admin-token` flag to the admin commands might be more manageable.

**Comment by dopey** [comment]
Ah, ok, that makes sense. If the cert/key are in a yubikey, then the admin middleware won't be able to correctly generate the necessary token.

We've created two new enhancements alongside this bug report - 
1. https://github.com/smallstep/cli/issues/863 -- This will enable users to pass in a fully formed x5c token to authenticate to the admin API.
2. https://github.com/smallstep/cli/issues/862 -- This will enable users to generate a token using KMS.

Thanks for bringing the issue to our attention @miguelangel-nubla, and for explaining the use case.

**Comment by miguelangel-nubla** [comment]
Great, the token approach seems a good workaround until kms support is everywhere.

> Some time ago I created [this script](https://gist.github.com/maraino/f73b334e475caa1a0f34a52c501fede7) that you should be able to use with a YubiKey too.

will definitely try that, thanks.

> Thanks for bringing the issue to our attention @miguelangel-nubla, and for explaining the use case.

Thanks for making this open-source.

**Comment by maraino** [comment]
@miguelangel-nubla there's no need for the script, in `master` we now have support for `--kms` with `step ca token`. But adding it to the admin client it's not straightforward as adding an `--admin-token` flag, but I think we are close to solve it.
